### ISSUE TITLE
[Box] Padding concise prop interface

### ISSUE THREAD

**Issue by Flufd** [Shopify/polaris-react-archive#7791]
## Feature request summary
Support an interface for clearly and concisely setting padding props on `<Box />`. 

## Rationale
Applying padding via the current props for inline/block start/end is okay when we want to individually apply styles to each of the 4 values. With the addition of responsive props this become 16 values for padding alone. 

```tsx
<Box
  paddingBlockStart={{
    sm: "1",
    md: "3",
    lg: "3",
    xl: "12"
  }}
  paddingBlockEnd={{
    sm: "1",
    md: "3",
    lg: "3",
    xl: "12"
  }}
  paddingInlineStart={{
    sm: "2",
    md: "4",
    lg: "8",
    xl: "12"
  }}
  paddingInlineEnd={{
    sm: "2",
    md: "4",
    lg: "8",
    xl: "13"
  }}
>
  <Placeholder label="Content inside a box" />
</Box>
```

I believe this could become confusing/noisy when trying to work on a layout single breakpoint, because the values for the same breakpoint are spread out over the interface of the prop. 

Further, we have no way to set both inline start and end, or block start and end padding at the same time. This is something that would happen very frequently and is possible in CSS with `padding: 5rem 3rem` `padding-inline: 5rem` for example, but we must apply the same value to both props.

My suggestion is an "inverted" interface that still allows for setting each level of padding with either a single token value, or a more specific set of values for each element at that "layer"

```
breakpoint => inline/block => start/end
```

The above code which has duplicated values for each start/end becomes:
```tsx
<Box
  padding={{
    sm: { inline: "2", block: "1" },
    md: { inline: "4", block: "3" },
    lg: { inline: "8", block: "3" },
    xl: { inline: "12", block: "12" }
  }}
>
  <Placeholder label="Content inside a box" />
</Box>
```

... and if we needed to go deeper in setting the inline/block start/end, or shallower, just setting the whole padding for a single breakpoint, we can still do this.
```tsx
<Box
  padding={{
    sm: { inline: "2", block: "1" },
    md: {
      inline: { start: "2", end: "4" },
      block: { start: "4", end: "4" }
    },
    lg: { inline: "8", block: "3" },
    xl: "12",
  }}
>
```

This would make the padding prop interface clearer by grouping related values together.

**Comment by kyledurand** [comment]
The 1:1 prop matching has gotten a bit out of hand and I'm a fan of how this looks. We've been thinking about grouping and might try to prototype something in the near future. Thanks for creating this issue ❤️

**Comment by Flufd** [comment]
I took a rough look at how this might be implemented, and wanted to drop it here. I wanted to build more of a concrete PR, but don't have the time right now. 

https://github.com/Shopify/polaris/pull/7856

Something I discovered when trying to implement this with the `responsive-props` mixin, is that the value supplied to `padding` on the box, is always overridden by the more specific `paddingInlineBlock` etc properties, which will be populated with the fallback value. The `padding` on the box is not required. Is that something the team is aware of?

<img width="1556" alt="image" src="https://user-images.githubusercontent.com/3304890/206271993-3f160636-3108-4b27-9b53-91e7cf24b25d.png">

I tried to add an intermediate level of `padding-block` and `padding-inline`  responsive props, but could not get the inheritance/fallback of the CSS properties to work correctly. The longhand values from the lowest breakpoint would always override the values being set at higher breakpoints. Falling back to the previous breakpoint values is always preferred to the more shorthand (`padding`) values at the current breakpoint. But the specificity of the previous breakpoint's values is not passed through in the fallback. This is not an issue as you have it now with one level of fallbacks, but I can't get it to work with more than one.

Instead in this example PR, I manually calculate each of the 20 values, that is, each of the four sides, for each of the 5 breakpoints, based on the input in javascript, and apply the individual variables to the box.
